tests/CachedServiceAccountJwtRetrieverTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace KeycloakAuthGuard\Tests;
+
+use Illuminate\Cache\Repository;
+use Illuminate\Http\Client\RequestException;
+use Illuminate\Support\Facades\Http;
+use KeycloakAuthGuard\Services\CachedServiceAccountJwtRetriever;
+use KeycloakAuthGuard\Services\ServiceAccountJwtRetriever;
+use Psr\SimpleCache\InvalidArgumentException;
+
+class CachedServiceAccountJwtRetrieverTest extends TestCase
+{
+    /**
+     * @throws RequestException
+     * @throws InvalidArgumentException
+     */
+    public function test_receiving_of_service_account_jwt()
+    {
+        Http::fake(fn () => Http::response($this->responseWithJwt()));
+
+        /** @var Repository $cacheStorage */
+        $cacheStorage = app('cache')->store('array');
+        $retriever = new CachedServiceAccountJwtRetriever(
+            new ServiceAccountJwtRetriever('', ''),
+            $cacheStorage
+        );
+        $jwt = $retriever->getJwt();
+
+        $this->assertEquals($this->responseWithJwt()['access_token'], $jwt);
+        $this->assertTrue($cacheStorage->has($retriever->getCacheKey()));
+        $this->assertEquals($cacheStorage->get($retriever->getCacheKey()), $jwt);
```

Comment:
**[non-blocker]** I would also assert that the retriever only queried Keycloak once using `Http::assertSentCount(1)`.

```suggestion
        $this->assertEquals($cacheStorage->get($retriever->getCacheKey()), $jwt);
        Http::assertSentCount(1)   
```